apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplateCardPreview/TemplateCardPreview.tsx

```diff
@@ -0,0 +1,135 @@
+} from '../../../../../__generated__/GetJourney'
+import { ThemeMode, ThemeName } from '../../../../../__generated__/globalTypes'
+import { CardWrapper } from '../../../CardPreview/CardList/CardWrapper'
+import { VideoWrapper } from '../../../Editor/Canvas/VideoWrapper'
+import { FramePortal } from '../../../FramePortal'
+
+SwiperCore.use([Mousewheel, A11y])
+
+interface TemplateCardPreviewProps {
+  steps?: Array<TreeBlock<StepBlock>>
+}
+
+interface TemplateCardPreviewItemProps {
+  step: TreeBlock<StepBlock>
+}
+
+export function TemplateCardPreview({
+  steps
+}: TemplateCardPreviewProps): ReactElement {
+  const smUp = useMediaQuery((theme: Theme) => theme?.breakpoints?.up('sm'))
+
+  function TemplateCardPreviewItem({
+    step
+  }: TemplateCardPreviewItemProps): ReactElement {
+    const { journey } = useJourney()
+    const { rtl, locale } = getJourneyRTL(journey)
+    const cardBlock = step.children.find(
+      (child) => child.__typename === 'CardBlock'
+    ) as TreeBlock<CardBlock>
+    return (
+      <Box
+        sx={{
+          display: 'flex',
+          flexDirection: 'column',
+          position: 'relative',
+          width: smUp ? 240 : 177,
+          height: smUp ? 380 : 280
+        }}
+      >
+        <Box
+          sx={{
+            transform: smUp ? 'scale(0.6)' : 'scale(0.4)',
+            transformOrigin: smUp ? 'top left' : '22.5% top'
+          }}
+        >
+          <Box
+            sx={{
+              position: 'absolute',
+              display: 'block',
+              width: smUp ? 405 : 445,
+              height: '100%',
+              zIndex: 2,
+              cursor: 'grab'
+            }}
+          />
+          <FramePortal
+            width={smUp ? 405 : 445}
+            height={smUp ? 633 : 698}
+            dir={rtl ? 'rtl' : 'ltr'}
+          >
+            <ThemeProvider
+              themeName={cardBlock?.themeName ?? ThemeName.base}
+              themeMode={cardBlock?.themeMode ?? ThemeMode.dark}
+              rtl={rtl}
+              locale={locale}
+            >
+              <Box
+                sx={{
+                  height: '100%',
+                  borderRadius: 4
+                }}
+              >
+                <BlockRenderer
+                  block={step}
+                  wrappers={{
+                    VideoWrapper,
+                    CardWrapper
+                  }}
+                />
+              </Box>
+            </ThemeProvider>
+          </FramePortal>
+        </Box>
+      </Box>
+    )
+  }
+
+  return (
+    <Swiper
+      freeMode
+      slidesPerView="auto"
+      spaceBetween={smUp ? 28 : 12}
+      slidesOffsetAfter={smUp ? 40 : 70}
+      mousewheel
+      style={{
+        marginLeft: smUp ? '-32px' : '-44px',
+        marginRight: smUp ? '-36px' : '-44px',
+        paddingLeft: smUp ? '32px' : '20px',
+        zIndex: 2
+      }}
+    >
+      {steps?.map((step) => (
+        <SwiperSlide
+          data-testid="swiper-container"
```

Comment:
This is the new pattern for component test ids. Also this is the slide, not the container
```suggestion
          data-testid="templateCardsSwiperSlide"
```